flink-formats/flink-json/src/test/java/org/apache/flink/formats/json/JsonRowDataSerDeSchemaTest.java

```diff
@@ -562,6 +579,56 @@ public void testSerializationMapNullKey() throws Exception {
         assertEquals(expectResult3, new String(actual3));
     }
 
+    @Test
+    public void testSerializationDecimalEncode() throws Exception {
+        BigDecimal decimal1 = new BigDecimal("123.456789");
+        BigDecimal decimal2 = new BigDecimal("454621864049246170");
+        BigDecimal decimal3 = new BigDecimal("0.000000027");
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.configure(JsonGenerator.Feature.WRITE_BIGDECIMAL_AS_PLAIN, true);
+
+        ObjectNode root = objectMapper.createObjectNode();
+        root.put("decimal1", decimal1);
+        root.put("decimal2", decimal2);
+        root.put("decimal3", decimal3);
+
+        byte[] serializedJson = objectMapper.writeValueAsBytes(root);
```

Comment:
I think we can manually construct the serialized JSON string.

Reply:
Modified test case to use explicit JSON strings.